File: back_end/fruitsapp_backend/Controllers/UserController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using fruitsapp_backend.Models;
using fruitsapp_backend.Models.DTOs;
using fruitsapp_backend.Service.Interfaces;
using Microsoft.AspNetCore.Mvc;

// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860

namespace fruitsapp_backend.Controllers
{
    [Route("api/[controller]")]
    public class UserController : Controller
    {
        private readonly IUserService _userService;

        public UserController(IUserService userService)
        {
            _userService = userService;
        }
        // GET: api/values
        [HttpGet]
        [Route("get-list-user")]
        public async Task<IActionResult> GetAllUser([FromQuery] int pageNumber, int pageSize)
        {
            try
            {
                var result = await _userService.GetListAsync(pageNumber,pageSize);


                return Ok(new { status = "success", message = "Get list successfully!", user = result });

            }
            catch (ArgumentException e)
            {
                // Bắt lỗi ArgumentException và trả về phản hồi BadRequest với thông điệp lỗi
                return BadRequest(new { status = "error", message = e.Message });
            }
            catch (Exception e)
            {
                return StatusCode(500, new { status = "error", message = "An error occurred while processing your request", error = e.Message });
            }
        }
        
        [HttpPost]
        [Route("disable-acounnt-user")]
        public async Task<IActionResult> DisableUser([FromBody] DisableUserRequest request )
        {
            try
            {
                var result = await _userService.DisableUserAsync(request.userId, request.isDisable);

                return Ok(new { status = "success", message = "Disable successfully!"});

            }
            catch (ArgumentException e)
            {
                // Bắt lỗi ArgumentException và trả về phản hồi BadRequest với thông điệp lỗi
                return BadRequest(new { status = "error", message = e.Message });
            }
            catch (Exception e)
            {
                return StatusCode(500, new { status = "error", message = "An error occurred while processing your request", error = e.Message });
            }
        }

        [HttpGet]
        [Route("details-infor-user/{userId}")]
        public async Task<IActionResult> DetailsUser([FromRoute]int userId)
        {
            try
            {
                var result = await _userService.DetailsUserAsync(userId);

                return Ok(new { status = "success", message = "Get user successfully!", user = result});

            }
            catch (ArgumentException e)
            {
                // Bắt lỗi ArgumentException và trả về phản hồi BadRequest với thông điệp lỗi
                return BadRequest(new { status = "error", message = e.Message });
            }
            catch (Exception e)
            {
                return StatusCode(500, new { status = "error", message = "An error occurred while processing your request", error = e.Message });
            }
        }

        [HttpDelete]
        [Route("delete-data-user")]
        public async Task<IActionResult> DeleteUser([FromForm] List<int> userId)
        {
            try
            {
                var result = await _userService.DeleteUserAsync(userId);

                return Ok(new { status = "success", message = "Deleted successfully!"});

            }
            catch (ArgumentException e)
            {
                // Bắt lỗi ArgumentException và trả về phản hồi BadRequest với thông điệp lỗi
                return BadRequest(new { status = "error", message = e.Message });
            }
            catch (Exception e)
            {
                return StatusCode(500, new { status = "error", message = "An error occurred while processing your request", error = e.Message });
            }
        }




    }
}

File: font_end/fruitsapp_admin/src/Controllers/userController.js
import UserRequest from "API/UserRequest/userRequest";
import User from "Models/User";
import ErrorHandler from "Utils/errorHandler";

const UserController = {
    getAllUser: async (pageNumber, pageSize) => {
        try {
            const response = await UserRequest.getAllUser(pageNumber, pageSize);
            const result = response.data;
            if (result.status === 'success') {
                // Chuyển đổi mỗi object trong danh sách thành một instance của User
                const users = result.user.map(u => new User(
                    u.id, 
                    u.role_id, 
                    u.name, 
                    u.avatar_url, 
                    u.email, 
                    u.phone, 
                    u.create_at, 
                    u.update_at, 
                    u.disable, 
                    u.role
                ));
                return { success: true, user: users, message: result.message };
            } else {
                return { success: false, message: result.message };
            }
        } catch (error) {
            return ErrorHandler.handle(error);
        }
    },
    disableUser: async (userId, isDisable) => {
        try {
            const response = await UserRequest.disableUser(userId, isDisable);
            const result = response.data;
            if (result.status === 'success') {
                return { success: true,  message: result.message };
            } else {
                return { success: false, message: result.message };
            }
        } catch (error) {
            return ErrorHandler.handle(error);
        }
    },
    detailsUser: async (userId) => {
        try {
            const response = await UserRequest.detailsUser(userId);
            const result = response.data;
            if (result.status === 'success') {
                const userData = new User(
                    result.user.id,
                    result.user.role_id,
                    result.user.name,
                    result.user.avatar_url,
                    result.user.email,
                    result.user.phone,
                    result.user.create_at,
                    result.user.update_at,
                    result.user.disable,
                    result.user.role);
                return { success: true, user: userData,  message: result.message };
            } else {
                return { success: false, message: result.message };
            }
        } catch (error) {
            return ErrorHandler.handle(error);
        }
    },
    editUser: async (userId) => {
        try {
            const response = await UserRequest.editUser(userId);
            const result = response.data;
            if (result.status === 'success') {
                const userData = new User(
                    result.data.id,
                    result.data.role_id,
                    result.data.name,
                    result.data.avatar_url,
                    result.data.email,
                    result.data.phone,
                    result.data.create_at,
                    result.data.update_at,
                    result.data.disable,
                    result.data.role);
                return { success: true, user: userData,  message: result.message };
            } else {
                return { success: false, message: result.message };
            }
        } catch (error) {
            return ErrorHandler.handle(error);
        }
    },
   
    
    
}

export default UserController;